specifyweb/frontend/js_src/lib/components/DataModel/resourceApi.ts

```diff
@@ -746,14 +747,21 @@ export const ResourceBase = Backbone.Model.extend({
     )
       return this;
     else if (this._fetch) return this._fetch;
-    else
-      return (this._fetch = Backbone.Model.prototype.fetch
-        .call(this, options)
-        .then(() => {
+    else {
+      const fetchCallback = () =>
+        Backbone.Model.prototype.fetch.call(this, options).then(() => {
           this._fetch = null;
           // BUG: consider doing this.needsSaved=false here
           return this;
-        }));
+        });
+      if (options === undefined || options.strict)
+        return (this._fetch = fetchCallback);
```

Comment:
shouldn't this be a function call instead?

```suggestion
        return (this._fetch = fetchCallback());
```

we don't have any type-checking in this file since there would be a few thousand errors)
till https://github.com/specify/specify7/issues/4286